### PR TITLE
Update doc to state that day partition transform produces data type

### PR DESCRIPTION
**Pull request by waterlx**
This PR is to fix the doc for #279,  to reflect the change made by #424

### CODE REVIEW

File: site/docs/spec.md [hunk 1]
```diff
@@ -139,7 +139,7 @@ Partition specs capture the transform from table data to partition values. This
 | **`truncate[W]`** | Value truncated to width `W` (see below)                     | `int`, `long`, `decimal`, `string`                                                                        | Source type |
 | **`year`**        | Extract a date or timestamp year, as years from 1970         | `date`, `timestamp(tz)`                                                                                   | `int`       |
 | **`month`**       | Extract a date or timestamp month, as months from 1970-01-01 | `date`, `timestamp(tz)`                                                                                   | `int`       |
-| **`day`**         | Extract a date or timestamp day, as days from 1970-01-01     | `date`, `timestamp(tz)`                                                                                   | `int`       |
+| **`day`**         | Extract a date or timestamp day, as date                     | `date`, `timestamp(tz)`                                                                                   | `date`      |
```
Comment (rdblue) [inline site/docs/spec.md]: This should change the type but not description.

Comment (waterlx) [reply site/docs/spec.md]: Comment addressed. Please review, thanks!

Comment (waterlx) [reply site/docs/spec.md]: @rdblue Comment addressed, Only "Return type" is changed. Please review at your most convenience, thanks!

Comment (waterlx) [reply site/docs/spec.md]: @rdblue Would you please help to review this change at your most convenience?

Comment (waterlx) [reply site/docs/spec.md]: Hi @rdblue Would you please help to review the change at your most convenience ? The comment addressed.

**Pull request merged**